decodestrings.py

```diff
@@ -0,0 +1,38 @@
+// Time Complexity : Solution incomplete
+// Space Complexity : Solution incomplete
+// Did this code successfully run on Leetcode : No
+// Any problem you faced while coding this : Yes, couldn't completely solve the question
+
+
+// Your code here along with comments explaining your approach
+
+class Solution:
+    def decodeString(self, s: str) -> str:
+        num = 0
+        index = 0
+        decoded = ""
+        intermediate = ""
+        return self.helper(s, index, num, decoded, intermediate)
+    
+    def helper(self, s, index, num, decoded, intermediate):
+        
+        if index == len(s):
+            return decoded
+        
+        if s[index].isdigit():
+            num = num*10 + int(s[index])
+            
+        if s[index].isalpha():
+            intermediate += s[index]
+            
+        if s[index] == '[':
+            num = 0
+            intermediate = ""
```

Comment:
when you encounter the '[' you set it empty string and num = 0 without even processing 
then when it reaches the ']' you should return until all the elements are processed in the brackets but then the condition terms only when index reaches the full length of the string. Let me know if you need more help.